hack/build-archive.sh

```diff
@@ -2,10 +2,24 @@
 
 cd data
 
-# RHEL tar doesn't include --sort=name, so have to use find
+# Normalise the archive accross different OS. We need consistent results in CI and in . Known issues:
+#  * RHEL `tar` doesn't include `--sort=name`, so have to use `find`
+#  * MacOS sets GUID different to 0 even if you sepecify `--group=root` in `tar`,
+#    so we have to explicitly set GUID to 0
+#  * MacOS sets different file mode for symlinks, so we have to explicitly set permissions for them
 
-find -not -path ./archive.tgz -and -not -path ./script.sh -print0 \
+TMP_TAR=$(mktemp)
+TAR_COMMON_OPTIONS="--no-recursion --null --files-from=- --owner=root --group=root --owner=0 --group=0 --mtime=@1546300800"
```

Comment:
wouldn't leaving `--owner=0 --group=0` work for all platforms?

Reply:
Tar stores [`uid` and `gid` as well as `uname` and `gname`](https://www.gnu.org/software/tar/manual/html_node/Standard.html). Only having `--owner=0 --group=0` without `--owner=root --group=root` will not work as expected on macOS: I tried that and got `wheel` in the `gname` field.

Despite the fact that I'm using GNU tar on macOS.